### PR TITLE
Auth Strategies

### LINKED ISSUE #66: Token Authentication
Should we change https://github.com/stormpath/stormpath-framework-spec/blob/master/cookie-authentication.md so it refers to Token Authentication as well? Or do we need to define what token authentication is, and a separate set of responses.

### PR DESCRIPTION
**Pull request by edjiang**
Added details about Bearer Tokens and Basic Auth to the spec. This is what we are already doing in Express and Laravel (although not sure about basic auth in Laravel). 

Fixes #66

### CODE REVIEW

File: user-context.md [hunk 1]
```diff
@@ -20,6 +20,10 @@ web:
     uri: "/me"
 ```
 
+### Authentication
+
+This endpoint requires authentication, and accepts all forms of authentication.
```
Comment (nbarbettini) [inline user-context.md:25]: How would Basic work here? API authentication?

Comment (edjiang) [inline user-context.md:25]: API credential Basic Auth is always associated with a user, so you'd be authenticated against that user. 

So Basic Auth as stated in https://github.com/stormpath/stormpath-framework-spec/pull/75/files#diff-940526024ff9cd3b0d3e9eae41005349R48

Comment (edjiang) [inline user-context.md:25]: I'm fine with this only accepting OAuth tokens as well, but wrote it in this way to get started. `/me` isn't that useful to be honest; I think most people would override it when they're starting to build their own application.

Comment (robertjd) [inline user-context.md:25]: As written, this makes sense to me.  Accept all forms of authentication, as this route will always have a response that is tied to the user context, and all of our authentication means do identify a user.

File: authenticators.md [hunk 2]
```diff
@@ -27,4 +27,28 @@ The default set of functions that we want to provide are:
...
+
+The framework integration should be able to read an access token presented in a Bearer Authentication scheme. As an example, this is a HTTP header that looks like this:
+
+```
+Authorization: Bearer <OAuth2AccessToken>
+```
+
+**Basic Authentication**
+
+The framework integration should be able to read API keys presented in Basic Authentication form. As an example, this is a HTTP header that looks like this:
```
Comment (robertjd) [inline authenticators.md]: Please add these notes: that these are API Keys that are for an Account resource (not Tenant API Keys)

File: authenticators.md [hunk 3]
```diff
@@ -27,4 +27,28 @@ The default set of functions that we want to provide are:
...
+**Cookie Authentication**
+
+The framework integration should have the ability to accept an OAuth access or refresh token in the cookies as detailed in the [cookie authentication](cookie-authentication.md) document.
+
+**Bearer Authentication**
+
+The framework integration should be able to read an access token presented in a Bearer Authentication scheme. As an example, this is a HTTP header that looks like this:
+
+```
+Authorization: Bearer <OAuth2AccessToken>
```
Comment (robertjd) [inline authenticators.md:45]: Please add these notes: that these access tokens can be created from the password grant flow, or the client credential flow.

**Comment by robertjd** [comment]
Looks good, thanks!  It does need some more context to be complete, please see my comments.

File: authenticators.md (hunk 2 above)
Comment (edjiang) [inline authenticators.md]: Tenant API keys are also linked to an account resource, correct? And technically you could map a Stormpath Admin directory to an application...

I could clarify to say that they're for a user mapped to the directory, but that's always been pretty implicit.

Comment (robertjd) [inline authenticators.md]: The implementation of our client credential workflow requires API Keys are that are a resource of an account, and that account must be enabled and accessible by the application the was used during the issuing of the token.

**Comment by edjiang** [comment]
Fixed!

**Pull request merged**